packages/editor/src/client/panel/imageSelectPanel.tsx

```diff
@@ -169,3 +176,9 @@ export function ImageSelectPanel({onClose, onSelect}: ImageSelectPanelProps) {
     </>
   )
 }
+
+const CheckedPermissionComponent = createCheckedPermissionComponent(
```

Comment:
Seems very repetitive, could we find a more generic solution/approach?

Reply:
This is already the generic solution. Instead of having to make the check we use a HoC generator here.